sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/Transforms.java

```diff
@@ -0,0 +1,305 @@
+
+            // add page Info
+            PageMetadata pageMetadata = getPageInfo(readResultItem);
+            PageRange pageRange = null;
+            DocumentResult documentResultItem = documentResult.get(i);
+            List<Integer> receiptPageRange = documentResultItem.getPageRange();
+            if (receiptPageRange.size() == 2) {
+                pageRange = new PageRange(receiptPageRange.get(0), receiptPageRange.get(1));
+            }
+            ExtractedReceipt extractedReceiptItem = new ExtractedReceipt(pageMetadata, pageRange);
+
+            // add receipt fields
+            documentResultItem.getFields().forEach((key, fieldValue) -> {
+                switch (key) {
+                    case "ReceiptType":
+                        extractedReceiptItem.setReceiptType(new ReceiptType(fieldValue.getValueString(),
+                            fieldValue.getConfidence()));
+                        break;
+                    case "MerchantName":
+                        extractedReceiptItem.setMerchantName(setFieldValue(fieldValue, readResults, includeTextDetails));
+                        break;
+                    case "MerchantAddress":
+                        extractedReceiptItem.setMerchantAddress(setFieldValue(fieldValue, readResults, includeTextDetails));
+                        break;
+                    case "MerchantPhoneNumber":
+                        extractedReceiptItem.setMerchantPhoneNumber(setFieldValue(fieldValue, readResults, includeTextDetails));
+                        break;
+                    case "Subtotal":
+                        extractedReceiptItem.setSubtotal(setFieldValue(fieldValue, readResults, includeTextDetails));
+                        break;
+                    case "Tax":
+                        extractedReceiptItem.setTax(setFieldValue(fieldValue, readResults, includeTextDetails));
+                        break;
+                    case "Tip":
+                        extractedReceiptItem.setTip(setFieldValue(fieldValue, readResults, includeTextDetails));
+                        break;
+                    case "Total":
+                        extractedReceiptItem.setTotal(setFieldValue(fieldValue, readResults, includeTextDetails));
+                        break;
+                    case "TransactionDate":
+                        extractedReceiptItem.setTransactionDate(setFieldValue(fieldValue, readResults, includeTextDetails));
+                        break;
+                    case "TransactionTime":
+                        extractedReceiptItem.setTransactionTime(setFieldValue(fieldValue, readResults, includeTextDetails));
+                        break;
+                    case "Items":
+                        extractedReceiptItem.setReceiptItems(toReceiptItems(fieldValue.getValueArray(), readResults, includeTextDetails));
+                        break;
+                    default:
+                        break;
```

Comment:
Instead of silently skipping over unknown keys, if the known set of keys are limited and new keys are not added frequently, this should throw an exception when the key is unknown. And when a new key is added, it should be accompanied by a service version update too right?

Reply:
The leftover keys would be put in a `Map<String, FieldValue> extractedFields` to still be able to capture them and not ignore.